test/weights.jl

```diff
@@ -447,6 +460,27 @@ end
     @test round(mean(Union{Int,Missing}[1,2], weights([1,2])), digits=3) ≈ 1.667
 end
 
+@testset "Sum, mean and quantiles for unit weights" begin
+    wt = uweights(Float64, 3)
+
+    @test sum([1.0, 2.0, 3.0], wt) ≈ 6.0
+    @test mean([1.0, 2.0, 3.0], wt) ≈ 2.0
+
+    @test sum(a, wt, 1) ≈ sum(a, dims=1)
+    @test sum(a, wt, 2) ≈ sum(a, dims=2)
+    @test sum(a, wt, 3) ≈ sum(a, dims=3)
+
+    @test mean(a, wt, dims=1) ≈ mean(a, dims=1)
+    @test mean(a, wt, dims=2) ≈ mean(a, dims=2)
+    @test mean(a, wt, dims=3) ≈ mean(a, dims=3)
+
+    @test_throws DimensionMismatch sum(a, wt, 4)
+    @test_throws DimensionMismatch mean(a, wt, dims=4)
+
+    @test quantile([1.0, 4.0, 6.0, 8.0, 10.0], uweights(5), [0.5]) ≈ [6.0]
```

Comment:
```suggestion
    @test quantile([1.0, 4.0, 6.0, 8.0, 10.0], uweights(5), 0.5) ≈ 6.0
```

Reply:
I used an array here on purpose (because the next test doesn't), but I'm open to changing it if you prefer. We could actually only keep the test of `median`, since it calls `quantile(x, w, [0.5]) `.

Reply:
Probably better test both forms for both methods then. You never know where bugs can happen.